src/routers/semester.ts

```diff
@@ -1,24 +1,53 @@
 import express from 'express';
+import { create, read, update, del } from '../operators/semesterOperations';
 
 const semesterRouter = express.Router();
 
-semesterRouter.get('/', (req, res) => {
-	console.log('Get Semester');
-	res.send('Get semesterRouter');
+semesterRouter.get('/', async (req, res) => {
+	try {
+		console.log('Get Semester');
+		const semester = await read(req.body.id);
+		const resData = {
+			...semester,
+			courses: Array.from(semester?.courses || []),
+		};
+		res.send(resData);
+	} catch (err) {
+		console.log(err);
+		res.send(err);
+	}
 });
 
-semesterRouter.post('/', (req, res) => {
-	console.log('Post Semester');
-	res.send('Post semesterRouter');
+semesterRouter.put('/', async (req, res) => {
+	try {
+		await create(req.body);
```

Comment:
We will want to make sure we create an id for the semester in the backend, since we want to control the ID.

Probably using something like uuid https://www.npmjs.com/package/uuid (v4 random uuid)

Reply:
cool! I will look into this!